Src/IronPython/Compiler/Ast/PythonNameBinder.cs

```diff
@@ -813,28 +813,10 @@ internal class PythonNameBinder : PythonWalker {
             node.Parent = _currentScope;
 
             if (node.Target is NameExpression nameExpr && nameExpr.Name == "super" && _currentScope is FunctionDefinition func) {
+                _currentScope.Reference("__class__");
```

Comment:
Should this go inside the if?

Reply:
Yes, the free variable should only be created when either `super` or `__class__` is referenced in the class body.

Reply:
I guess I should have been more specific. I meant the if below. 😄

Reply:
I see. I think you are right, though this is from previous code, it was like this before the "hack", not sure if we should touch it :)

Reply:
I checked this against CPython and the way you have it is correct. Calling `super` with arguments still adds `__class__` to the scope:
```Python
class test:
    def test(self):
        assert "__class__" in dir()
        super(test, self).__init__() # calling super with arguments still adds __class__ to the local scope

test().test()
```